src/main/java/org/spongepowered/api/text/channel/MessageChannel.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.channel;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableSet;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.channel.impl.DelegateMutableMessageChannel;
+import org.spongepowered.api.text.channel.type.CombinedMessageChannel;
+import org.spongepowered.api.text.channel.type.FixedMessageChannel;
+import org.spongepowered.api.text.channel.type.PermissionMessageChannel;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a channel that takes a message and transforms it for distribution
+ * to the members.
+ */
+public interface MessageChannel {
+
+    /**
+     * A channel with no members.
+     */
+    MessageChannel TO_NONE = ImmutableSet::of;
+    /**
+     * A channel with all online players as members.
+     */
+    MessageChannel TO_PLAYERS = () -> ImmutableSet.copyOf(Sponge.getGame().getServer().getOnlinePlayers());
+    /**
+     * A channel with the server console as a member.
+     */
+    MessageChannel TO_CONSOLE = () -> ImmutableSet.of(Sponge.getGame().getServer().getConsole());
+    /**
+     * A channel with all online players, as well as the server console, as members.
+     */
+    MessageChannel TO_ALL = () -> ImmutableSet.<MessageReceiver>builder()
```

Comment:
Since there is no `server.getRconSources()` this needs to be handled by the server itself (exchange the impl) or do you have an alternative suggestion?